source/plugins/go/src/oms.go

```diff
@@ -729,9 +734,12 @@ func flushKubeMonAgentEventRecords() {
 				}
 			}
 			if IsWindows == false && len(msgPackEntries) > 0 { //for linux, mdsd route
-				if IsAADMSIAuthMode == true && strings.HasPrefix(MdsdKubeMonAgentEventsTagName, MdsdOutputStreamIdTagPrefix) == false {
-					Log("Info::mdsd::obtaining output stream id for data type: %s", KubeMonAgentEventDataType)
-					MdsdKubeMonAgentEventsTagName = extension.GetInstance(FLBLogger, ContainerType).GetOutputStreamId(KubeMonAgentEventDataType)
+				if IsAADMSIAuthMode == true {
+					MdsdKubeMonAgentEventsTagName = getOutputStreamIdTag(KubeMonAgentEventDataType, MdsdKubeMonAgentEventsTagName, &MdsdKubeMonAgentEventsTagRefreshTracker)
+					if MdsdKubeMonAgentEventsTagName == "" {
+						Log("Warn::mdsd::skipping Microsoft-KubeMonAgentEvents stream since its opted out")
+						return
```

Comment:
does this one also needs return output.FLB_OK?

Reply:
> does this one also needs return output.FLB_OK or Retry?

no. we dont need either retry or return output.FLB_OK. this table just for config errors and this is not useful data we should keep re-trying.